packages/itwin/tree-widget/src/tree-widget-react/components/trees/common/components/TreeNodeVisibilityButton.tsx

```diff
@@ -29,16 +29,24 @@ export interface TreeItemVisibilityButtonProps {
   onVisibilityButtonClick: (node: PresentationHierarchyNode, state: TreeItemVisibilityButtonState["state"]) => void;
   /** Callback that should be used to determine current checkbox state. */
   getVisibilityButtonState: (node: PresentationHierarchyNode) => TreeItemVisibilityButtonState;
+  /**
+   * Indicates that space for this action button should be reserved, even when the action is not available.
+   * For nodes that don't support visibility, `<VisibilityAction reserveSpace />` renders:
+   *
+   * - Blank space when the action is used as an inline action. It's recommended to set this prop to keep all action buttons of the same kind vertically aligned.
+   * - Disabled menu item when the action is used as a menu action.
+   */
+  reserveSpace?: true;
```

Comment:
We have this elsewhere:
```ts
type VisibilityTreeRendererProps = BaseTreeRendererProps & TreeItemVisibilityButtonProps;
```

Which means `VisibilityTreeRendererProps` now also get the `reserveSpace` prop. I guess that's not intended? I think `VisibilityTreeRendererProps` should take specific properties from `TreeItemVisibilityButtonProps` that it needs with `Pick`.

Reply:
changed to using visibilityContext type.